source/common/http/filter_manager.cc

```diff
@@ -1120,7 +1120,7 @@ void FilterManager::encodeMetadata(ActiveStreamEncoderFilter* filter,
     // If the filter pointed by entry hasn't returned from encodeHeaders, stores newly added
     // metadata in case encodeHeaders returns StopAllIteration. The latter can happen when headers
     // callbacks generate new metadata.
-    if (!(*entry)->encode_headers_called_ || (*entry)->stoppedAll()) {
+    if (!(*entry)->processed_headers_ || (*entry)->stoppedAll()) {
```

Comment:
Sorry, can you explain why this is safe? Can `decodeHeaders, encodeMetadata` happen in a sequence for dual filters?

Reply:
dual filter -> encoder-decoder filters. 

The `processed_headers_` bit is on the wrapper for a filters. So an encoder-decoder filter would have both a encoder-wrapper and a decoder-wrapper which would each have their own seperate `processed_headers_` bit.

Reply:
Thanks.